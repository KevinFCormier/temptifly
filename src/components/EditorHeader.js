'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { Tab, Tabs, Checkbox } from 'carbon-components-react'
import '../scss/editor-header.scss'

class EditorHeader extends React.Component {
  static propTypes = {
    children: PropTypes.node,
    handleShowSecretChange: PropTypes.func,
    handleTabChange: PropTypes.func,
    i18n: PropTypes.func,
    otherYAMLTabs: PropTypes.array,
    showSecrets: PropTypes.bool,
    type: PropTypes.string.isRequired
  };

  constructor(props) {
    super(props)
  }

  render() {
    const { children, otherYAMLTabs = [], i18n } = this.props
    const editorToolbarTitle = i18n('editor.toolbar')
    const hasTabs = otherYAMLTabs.length > 0
    const classnames = classNames({
      'creation-view-yaml-header': true,
      hasTabs: hasTabs
    })
    return (
      <div className={classnames}>
        <div
          className="creation-view-yaml-header-toolbar"
          role="region"
          aria-label={editorToolbarTitle}
          id={editorToolbarTitle}
        >
          {children}
        </div>
        {hasTabs && (
          <div className="creation-view-yaml-header-tabs">
            {this.renderEditorTabs(otherYAMLTabs)}
            {this.renderShowSecrets()}
          </div>
        )}
      </div>
    )
  }

  renderEditorTabs = otherYAMLTabs => {
    const { type = 'unknown', handleTabChange } = this.props
    return (
      <Tabs selected={0} aria-label={'Select template'}>
        <Tab
          label={type}
          key="main"
          id="main"
          onClick={handleTabChange.bind(this, 0)}
        />
        {otherYAMLTabs.map(({ id }, inx) => {
          return (
            <Tab
              label={id}
              key={id}
              id={id}
              onClick={handleTabChange.bind(this, inx + 1)}
            />
          )
        })}
      </Tabs>
    )
  };

  renderShowSecrets = () => {
    const { showSecrets, handleShowSecretChange, i18n } = this.props
    return (
      <div className="creation-view-yaml-header-secrets">
        <Checkbox
          id="show-secrets"
          className="checkbox"
          hideLabel
          labelText=""
          checked={showSecrets}
          onChange={handleShowSecretChange}
        />
        <div>{i18n('editor.show.secrets')}</div>
      </div>
    )
  };
}

export default EditorHeader
