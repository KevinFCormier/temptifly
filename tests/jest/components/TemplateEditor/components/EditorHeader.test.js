'use strict'

import React from 'react'
import EditorHeader from '../../../../../src/components/EditorHeader'
import renderer from 'react-test-renderer'

const type = 'application'
const otherYAMLTabs = []
const handleTabChange = jest.fn

describe('EditorHeader component', () => {
  it('renders as expected', () => {
    const fn = jest.fn()
    const component = renderer.create(
      <EditorHeader
        otherYAMLTabs={otherYAMLTabs}
        handleTabChange={handleTabChange}
        type={type}
        i18n={fn}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
