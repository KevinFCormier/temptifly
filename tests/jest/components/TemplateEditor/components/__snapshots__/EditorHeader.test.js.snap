// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorHeader component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="creation-view-yaml-header"
  >
    <div
      class="creation-view-yaml-header-toolbar"
      role="region"
    />
  </div>
</DocumentFragment>
`;
